src/core/annotation.js

```diff
@@ -920,20 +935,98 @@ class TextWidgetAnnotation extends WidgetAnnotation {
       this.data.maxLen !== null;
   }
 
-  getOperatorList(evaluator, task, renderForms) {
-    if (renderForms || this.appearance) {
-      return super.getOperatorList(evaluator, task, renderForms);
-    }
+  getOperatorList(evaluator, task, renderForms, annotationStorage) {
+    const self = this;
+    return this.getAppearance(evaluator, task, annotationStorage).then(
+      content => {
+        if (renderForms || (self.appearance && content === "")) {
+          return super.getOperatorList(
+            evaluator,
+            task,
+            renderForms,
+            annotationStorage
+          );
+        }
 
-    const operatorList = new OperatorList();
+        const operatorList = new OperatorList();
+
+        // Even if there is an appearance stream, ignore it. Self is the
+        // behaviour used by Adobe Reader.
+        if (!self.data.defaultAppearance || content === "") {
+          return Promise.resolve(operatorList);
+        }
+
+        const matrix = [1, 0, 0, 1, 0, 0];
+        const bbox = [
+          0,
+          0,
+          self.data.rect[2] - self.data.rect[0],
+          self.data.rect[3] - self.data.rect[1],
+        ];
+
+        const transform = getTransformMatrix(self.data.rect, bbox, matrix);
+        operatorList.addOp(OPS.beginAnnotation, [
+          self.data.rect,
+          transform,
+          matrix,
+        ]);
+
+        const stream = new Stream(stringToBytes(content));
```

Comment:
If you import `StringStream` from https://github.com/mozilla/pdf.js/blob/master/src/core/stream.js, you can instead do
```suggestion
        const stream = new StringStream(content);
```